### ISSUE TITLE
Some objects from mods are inaccessible

### ISSUE THREAD

**Issue by DjWarmonger** [vcmi/vcmi#1841]
**Describe the bug**
RMG sometimes can construct a treasure pile containing modded object, which is unaccessible.
Not sure if or why it's specific only to objects from mods, but didn't notice it for regular objects.

**To Reproduce**
Install Tides of War, Antagarich Buring and maybe HotA to get objects from screenshots.

**Expected behavior**
Objects should always be accessible.

**Screenshots**

<img width="157" alt="Inaccessible Elemental Conflux" src="https://user-images.githubusercontent.com/2566990/229340414-0e71f953-c033-4b24-b457-08cc5ebed17b.png"><img width="135" alt="Inaccessible Gazebo" src="https://user-images.githubusercontent.com/2566990/229340416-e7060534-83bd-4e2c-9447-1d24a2a597c1.png"><img width="127" alt="Inaccessible Magic Lamp" src="https://user-images.githubusercontent.com/2566990/229340418-5b598707-38e2-4f97-8286-b04c8c43e1ed.png"><img width="114" alt="Inaccessible Sorceress Castle" src="https://user-images.githubusercontent.com/2566990/229340419-2ab76b66-9053-46d8-b9ad-175f4d897484.png">

**Version**
Beta

**Additional context**
 It is not a duplicate of Corpse bug: https://github.com/vcmi/vcmi/issues/847

**Comment by misiokles** [comment]
Sometimes editor passabilities can be 'misleading'.  I encountered many times that given object is accesible, but first we must visit object next to it, and return in the same way. It concerns also standard objects (which is not OH3 behaviour). Is it connected with masks management?

**Comment by DjWarmonger** [comment]
I noticed that all these objects have `"visitableFrom"` set to all surrounding tiles, even though it's incorrect. They shouldn't be visitable from tiles blocked by themselves.

Still, RMG should handle that on its own. Need more investigation.

**Comment by misiokles** [comment]
Most of them were taken (or inspired from here) https://heroes.thelazy.net/index.php/List_of_adventure_map_objects_(HotA)

**Comment by DjWarmonger** [comment]
Nope, for example Gazebo is wrong. Shouldn't be accessible from top.

**Comment by DjWarmonger** [comment]
Oh, I got it. RMG has special treatment fort "large" objects, that is objects which are not visitable from top. These unfortunatelly  are, and this is confusing.
So I call it incorrect mod content as well.
